js/client-teambuilder.js

```diff
@@ -112,6 +112,13 @@
 		curSet: null,
 		curSetLoc: 0,
 
+		// Instead of running a search after each character typed
+		// in the search bar we instead use a timer to only kick off
+		// a search 500ms after the last 'keyup' event
+		// (which is intended to approximate running a search only
+		//  after the user has 'finished' typing).
```

Comment:
Why? This makes the search feel laggy for no real benefit.

Reply:
I mean, the timer can definitely be adjusted?

The main concern @TheJetOU  raised in <<dev>> was the performance of having to parse through potentially a very large amount of teams each keystroke. Maybe its less of a concern now that we've optimized it to just be doing `splitFirst` and search names, but adding some sort of timer is a fairly standard search-as-you-type optimization?

Reply:
The perf isn't actually a problem. On the other hand, render perf is... You'd want to only render, say, the top 12 teams or so. For the rest, you'd only want to display them on scroll.

Reply:
(If "on scroll" is too hard, you can also just display a clickable "More" button.)